crates/codegen/src/compile.rs

```diff
@@ -2121,12 +2130,150 @@ impl Compiler {
 
     fn compile_try_star_statement(
         &mut self,
-        _body: &[Stmt],
-        _handlers: &[ExceptHandler],
-        _orelse: &[Stmt],
-        _finalbody: &[Stmt],
+        body: &[Stmt],
+        handlers: &[ExceptHandler],
+        orelse: &[Stmt],
+        finalbody: &[Stmt],
     ) -> CompileResult<()> {
-        Err(self.error(CodegenErrorType::NotImplementedYet))
+        let handler_block = self.new_block();
+        let finally_block = self.new_block();
+
+        if !finalbody.is_empty() {
```

Comment:
I'd rather have it split like CPython does it

https://github.com/python/cpython/blob/627894459a84be3488a1789919679c997056a03c/Python/compile.c#L3836-L3845

This makes it easier to maintain when looking for the diff between our impl and theirs